src/service-api.ts

```diff
@@ -19,24 +20,68 @@ const plugin: FastifyPluginAsync = async (fastify) => {
   // get current user
   fastify.get('/current', async ({ member }) => member);
 
-    // // get member
-    // fastify.get(
-    //   '/:id',
-    //   { schema: getOne },
-    //   async ({ category, id , log }) => {
-    //     const task = taskManager.createGetTask(category, id);
-    //     return runner.runSingle(task, log);
-    //   },
-    // );
-
-    // get members
-    fastify.get(
-      '/allcategories',
-      async ({ member, log }) => {
-        const task = taskManager.createGetAllTask(member);
+    // get category info
+  fastify.get<{ Params: {categoryId: string };}>(
+    '/category/info/:categoryId',
+    async ({ member, params: {categoryId}, log }) => {
+      const task = taskManager.createGetCategoryTask(member, categoryId);
+      return runner.runSingle(task, log);
+    },
+  );
+
+  // get types
+  fastify.get(
+    '/category/types',
+    async ({ member, log }) => {
+      const task = taskManager.createGetCategoryTypesTask(member);
+      return runner.runSingle(task, log);
+    },
+  );
+
+  // get categories of given type
+  fastify.get<{ Params: {typeId: string}; }>(
+    '/categories/:typeId',
+    async ({ member, params: {typeId}, log }) => {
+      const task = taskManager.createGetCategoriesByTypeTask(member, typeId);
+      return runner.runSingle(task, log);
+    },
+  );
+
+  //get category of an item
+  fastify.get<{ Params: {itemId: string };}>(
+    '/:itemId/categories',
+    async ({ member, params: {itemId}, log }) => {
+      const task = taskManager.createGetItemCategoryTask(member, itemId);
+      return runner.runSingle(task, log);
+    },
+  );
+
+  // get items in given category
+  fastify.get<{ Params: {categoryId: string };}>(
+    '/category/:categoryId',
+    async ({ member, params: {categoryId}, log }) => {
+      const task = taskManager.createGetItemsByCategoryTask(member, categoryId);
+      return runner.runSingle(task, log);
+    },
+  );
+
+    // insert item category
+  fastify.post<{ Params: { itemId: string }; Body: ItemCategory }>(
+    '/category/create/:itemId', { schema: create },
+    async ({ member, params: { itemId }, body, log }) => {
+      const task = taskManager.createCreateItemCategoryTask(member, body, itemId);
+      return runner.runSingle(task, log);
+    },
+  );   
+
+    // delete item category entry
+    fastify.post<{ Params: { itemId: string }; Body: ItemCategory }>(
+      '/category/delete/:itemId', { schema: create },
```

Comment:
Same as above with `/categories/:itemId`
And actually itemid is the item_category's id right? Maybe rename it for `id` only.

Reply:
For this one, I rename it to /item_category/:id, because we are removing an entry from the table item_category, and also we might need another endpoint to remove category from category table (in the future), I think it's better to name that one /categories/:categoryId